gallery_dl/extractor/deviantart.py

```diff
@@ -148,12 +155,23 @@ def deviations(self):
 
     def prepare(self, deviation):
         """Adjust the contents of a Deviation-object"""
+        alphabet = "0123456789abcdefghijklmnopqrstuvwxyz"
         if "index" not in deviation:
             try:
-                deviation["index"] = text.parse_int(
-                    deviation["url"].rpartition("-")[2])
+                if deviation["url"].startswith("https://sta.sh"):
+                    filename = deviation["content"]["src"].split("/")[5]
+                    deviation["index_base36"] = filename.partition("-")[0][1:]
+                    deviation["index"] = \
+                        util.bdecode(deviation["index_base36"], alphabet)
+                else:
+                    deviation["index"] = text.parse_int(
+                        deviation["url"].rpartition("-")[2])
             except KeyError:
                 deviation["index"] = 0
+                deviation["index_base36"] = "0"
+        if "index_base36" not in deviation:
+            deviation["index_base36"] = \
+                util.bencode(deviation["index"], alphabet)
```

Comment:
This is used to handle shared sta.sh deviations in status updates (see https://github.com/mikf/gallery-dl/issues/3539#issuecomment-1398259478). I moved the code to the parent extractor because it's also possible to extract embedded sta.sh deviations in journals, and I plan to add this feature in the future.